### DISCUSSION TITLE
Potential Queries for Virtuoso

### DISCUSSION THREAD

**Discussion by homework36** [DDMAL/linkedmusic-datalake#32] [General]
Hello,

To better work with Virtuoso and experiment with its search functionality, Van (@malajvan) and I are collecting potential search queries. These queries are intended to set up and test Virtuoso so that we can better assess if and how we want to implement it for the entire project. Below is a preliminary list of such queries that we have compiled so far.

- Facts:
    - (Original query request from Ich) Find all such composers in simssadb: born before 1500, wrote a piece in the Phrygian mode, and had a title containing "Ave Maria"
    - Find a composer who wrote pieces with lyrics in multiple languages
    
- Statistics (with our datasets plus wikidata):
    - Find the most frequent mode (or anything) used by [a specific composer] during 1550-1600 
    - Find the composer who wrote the most chants in the year of 1600
    - Compare the use of minor keys in solo instrument compositions (by percentage) in 1500-1550 and in 1550-1600
    
- Enhanced Search Functionality (later):
    - Search in different languages
    - Fuzzy search
    - Recommend similar results or suggest “Did you mean xxx” when the user entered "xxy"

We are currently looking for more input. If you have any ideas for queries we can use or you want to use with Virtuoso, we would love to know. We also welcome suggestions to create a more comprehensive query list.

Thank you very much!

**Comment by fujinaga** [comment]
Any queries that need information from more than one database would be good test queries.

"Find the composer who wrote the most chants in the year of 1600" doesn't make sense as most chants have no composers associated with them. Also dating of chants are difficult. At best, we can get the approximate date when the manuscript containing the chant was copied.
